File: src/test/java/com/github/jactor/web/menu/MenuFacadeTest.java
package com.github.jactor.web.menu;

import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;

import java.util.Collections;
import java.util.List;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

@DisplayName("A MenuFacade")
class MenuFacadeTest {

  @Test
  @DisplayName("should fail if the menu asked for is unknown")
  void willFailWhenMenuIsUnknown() {
    MenuItem menuItem = new MenuItem("name", "target", null, Collections.emptyList());
    MenuFacade menuFacadeToTest = new MenuFacade(List.of(
        new Menu("known", List.of(menuItem))
    ));

    assertThatIllegalArgumentException().isThrownBy(() -> menuFacadeToTest.fetchMenuItemsByName("unknown.menu"))
        .withMessageContaining("unknown.menu")
        .withMessageContaining("known.menu");
  }

  @Test
  @DisplayName("should find menu items for known Menu")
  void willFindKnownMenuItems() {
    MenuItem menuItem = new MenuItem("name", "target", null, Collections.emptyList());
    MenuFacade menuFacadeToTest = new MenuFacade(List.of(
        new Menu("known", List.of(menuItem))
    ));

    List<MenuItem> menuItems = menuFacadeToTest.fetchMenuItemsByName("known");

    Assertions.assertThat(menuItems).contains(menuItem);
  }
}

File: src/test/kotlin/com/github/jactor/web/menu/MenuFacadeTest.kt
package com.github.jactor.web.menu

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatIllegalArgumentException
import org.junit.jupiter.api.Test

internal class MenuFacadeTest {

    @Test
    fun `should fail if the menu asked for is unknown`() {
        val menuItem = MenuItem(itemName = "name", target = "target")
        val menuFacadeToTest = MenuFacade(
            listOf(
                Menu("known", mutableListOf(menuItem))
            )
        )

        assertThatIllegalArgumentException().isThrownBy { menuFacadeToTest.fetchMenuItemsByName("unknown.menu") }
            .withMessageContaining("unknown.menu")
            .withMessageContaining("known.menu")
    }

    @Test
    fun `should find menu items for known Menu`() {
        val menuItem = MenuItem(itemName = "name", target = "target")
        val menuFacadeToTest = MenuFacade(
            listOf(
                Menu("known", mutableListOf(menuItem))
            )
        )

        val menuItems = menuFacadeToTest.fetchMenuItemsByName("known")

        assertThat(menuItems).contains(menuItem)
    }
}
